testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/web/security/authentication/BasicAuthenticationMechanismPicketboxRemovedTestCase.java

```diff
@@ -0,0 +1,93 @@
+import org.jboss.arquillian.container.test.api.Deployment;
+import org.jboss.arquillian.container.test.api.RunAsClient;
+import org.jboss.arquillian.junit.Arquillian;
+import org.jboss.arquillian.test.api.ArquillianResource;
+import org.jboss.as.arquillian.api.ServerSetup;
+import org.jboss.as.test.integration.security.common.Utils;
+import org.jboss.as.test.integration.web.security.authentication.deployment.SecuredEJB;
+import org.jboss.as.test.integration.web.security.authentication.deployment.SecuredEJBServlet;
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.spec.WebArchive;
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import java.net.URL;
+
+/**
+ * Automated test for [ WFLY-10282 ]
+ * Test Enables elytron and removes security subsystem. After secured EJB is deployed and accessed using basic authorization
+ * test checks if no NullPointerException is thrown.
+ *
+ * Test is also related to the issue [ WFLY-10292 ]: after switching to elytron and removing picketbox subsystem DefaultJMSConnectionFactory is not found during server startup.
+ *
+ * @author Daniel Cihak
+ */
+@RunWith(Arquillian.class)
+@ServerSetup(BasicAuthMechanismServerSetupTask.class)
+public class BasicAuthenticationMechanismPicketboxRemovedTestCase {
+
+    private static final String EJB_SECURITY = "ejb-security";
+
+    @Deployment(name = EJB_SECURITY)
+    public static WebArchive appDeployment1() {
+        WebArchive war = ShrinkWrap.create(WebArchive.class, EJB_SECURITY + ".war");
+        war.addClasses(BasicAuthenticationMechanismPicketboxRemovedTestCase.class, SecuredEJB.class, SecuredEJBServlet.class);
+        war.setWebXML(Utils.getResource("org/jboss/as/test/integration/web/security/authentication/deployment/web.xml"));
+        war.addAsWebInfResource(Utils.getResource("org/jboss/as/test/integration/web/security/authentication/deployment/jboss-web.xml"), "jboss-web.xml");
+
+        return war;
+    }
+
+    /**
+     * Test checks if no NullPointerException is thrown after secured EJB is accessed via http call.
+     *
+     * @param url
+     * @throws Exception
+     */
+    @Test
+    @RunAsClient
+    public void test(@ArquillianResource URL url) throws Exception {
+        HttpResponse response;
+        HttpGet httpget = new HttpGet(url.toString() + EJB_SECURITY + "/SecuredEJBServlet");
+        String headerValue = java.util.Base64.getEncoder().encodeToString("user:password".getBytes());
+        httpget.addHeader("Authorization", "Basic " + headerValue);
+        String text;
+
+        try (CloseableHttpClient httpclient = HttpClients.createDefault()){
+            response = httpclient.execute(httpget);
+            text = Utils.getContent(response);
+            int statusCode = response.getStatusLine().getStatusCode();
+
+            Assert.assertEquals(401, statusCode);
+            Assert.assertFalse(text.contains("java.lang.NullPointerException"));
```

Comment:
This should not be here. Because the result is a 401 the error page writer has not been trigged, and this is not the correct way to write a test anyway. You assert correct behaviour, you don't check for a specific type of error in the resulting page.

In this case it means checking that the response contains the expected headers, and then sending a request with correct credentials and verifying that it works as expected. 

Even though the initial bug report the failure mode was a NPE that is not really relevant, the real issue is that this did not work, so your test needs to verify that it works, not it no longer throws a NPE.